### PR TITLE
Feature/expand with dst

### PR DESCRIPTION
**Pull request by Shylock-Hg**
Expand to specific dest vertex.

Require https://github.com/vesoft-inc/nebula-common/pull/543

### CODE REVIEW

File: src/storage/exec/StorageIterator.h [hunk 1]
```diff
@@ -114,6 +116,16 @@ class SingleEdgeIterator : public StorageIterator {
             return false;
         }
 
+        // check dst
+        if (dstId_.hasValue()) {
+            auto rawDstId = NebulaKeyUtils::getDstId(planContext_->vIdLen_, iter_->key());
+            // TODO optimize the compare
+            dstId_.value().resize(rawDstId.size());
+            if (dstId_.value() != rawDstId) {
```
Comment (critical27) [inline src/storage/exec/StorageIterator.h:124]: Why not just push down a filter?

Comment (Shylock-Hg) [reply src/storage/exec/StorageIterator.h:124]: So what's the filter expression?

File: src/storage/query/GetNeighborsProcessor.cpp [hunk 2]
```diff
@@ -61,39 +61,71 @@ void GetNeighborsProcessor::doProcess(const cpp2::GetNeighborsRequest& req) {
...
-            auto vId = row.values[0].getStr();
-
-            if (!NebulaKeyUtils::isValidVidLen(spaceVidLen_, vId)) {
-                LOG(ERROR) << "Space " << spaceId_ << ", vertex length invalid, "
-                           << " space vid len: " << spaceVidLen_ << ",  vid is " << vId;
-                pushResultCode(nebula::cpp2::ErrorCode::E_INVALID_VID, partId);
-                onFinished();
-                return;
-            }
+    if (!withDst_) {
```
Comment (critical27) [inline src/storage/query/GetNeighborsProcessor.cpp:64]: If we don't implement this feature by push down the filter. Maybe you could save a dstId in `PlanContext` or `EdgeContext`, and just pass it to `StorageIterator`, the template seems unnecessary.

But I sill prefer by adding a filter.

Comment (Shylock-Hg) [reply src/storage/query/GetNeighborsProcessor.cpp:64]: There are pair of <src,dst> for each edge scan, so I think it should passed by the StoragePlan::go method.

File: src/storage/exec/StorageIterator.h (hunk 1 above)
Comment (critical27) [reply src/storage/exec/StorageIterator.h:124]: `edge._dst == xxx`?

Comment (Shylock-Hg) [reply src/storage/exec/StorageIterator.h:124]: But the `xxx` is different for each vertex.It's not constant.

Comment (critical27) [reply src/storage/exec/StorageIterator.h:124]: I know exactly what you mean here. We need to design related query grammar. What if we need to specify rank as well, add one extra template? It seems unacceptable to me.

Comment (Shylock-Hg) [reply src/storage/exec/StorageIterator.h:124]: If you don't want the template parameter, I think could let the StoragePlan::go accept the Row type as the input.

**Pull request closed without merging**